### ISSUE TITLE
[feature request] Add an assertNotMatch function for std/testing/asserts.ts

### ISSUE THREAD

**Issue by xcatliu** [denoland/deno#6773]
There is an `assertMatch` function in `std/testing/asserts.ts`
Please considering to add an `assertNotMatch` function.
It'll be useful when testing regexp

**Comment by xcatliu** [comment]
If this idea is approved, I would like to send a pr for this.

**Comment by nayeemrmn** [comment]
You can use negative lookahead to negate a regex matcher, no?

**Comment by xcatliu** [comment]
Sometimes the input regex cannot be edited, for example I want to test a regex which belongs to a class:

```ts
import Pagic from './src/Pagic.ts';
import {  assertNotMatch } from 'https://deno.land/std/testing/asserts.ts';

Deno.test('Pagic.REGEXP', () => {
  assertNotMatch('foo/bar', Pagic.REGEXP_LAYOUT);
});
```

Both jest and mocha have the notMatch assertion in there built-in asserts:

- https://jestjs.io/docs/en/expect#expectnotstringmatchingstring--regexp
- https://nodejs.org/api/assert.html#assert_assert_doesnotmatch_string_regexp_message

**Comment by nayeemrmn** [comment]
Fair enough, I guess regexes can't be generally negated anyway.

**Comment by ebebbington** [comment]
It makes sense, we have `assertEquals` and `assertNotEquals`

**Comment by caspervonb** [comment]
ECMAScript's regex engine isn't the strongest one in terms of lookahead's and lookbehind's so this one makes sense IMHO.

**Comment by bartlomieju** [comment]
Closed in #6775
